packages/core/src/shared/lsp/utils/platform.ts

```diff
@@ -81,26 +86,121 @@ export async function validateNodeExe(nodePath: string[], lsp: string, args: str
     }
 }
 
+/**
+ * Gets proxy settings and certificates from VS Code
+ */
+export function getVSCodeSettings(): { proxyUrl?: string; certificatePath?: string } {
+    const result: { proxyUrl?: string; certificatePath?: string } = {}
+    const logger = getLogger('amazonqLsp')
+
+    try {
+        // Check if user already has NODE_EXTRA_CA_CERTS set
+        const userCerts = process.env.NODE_EXTRA_CA_CERTS
```

Comment:
not sure if this works correctly on mac -- see discussion in jetbrains/eclipse where macOS does not inherit the console environment due to how app launching works